File: ABC183/D.cpp
#include <bits/stdc++.h>
using namespace std;

int main() { 
    int N, W;
    cin >> N >> W;

    vector<long long> cusum(200001); 
    for(int i = 0; i < N; i++) {
        int s, t, p;
        cin >> s >> t >> p;

        cusum[s] += p;
        cusum[t] -= p;
    }

    for (int i = 0; i < cusum.size() - 1; i++) {
        cusum[i + 1] += cusum[i];
    }

    for (int i = 0; i < cusum.size(); i++) {
        if (cusum[i] > W) {
            cout << "No" << endl;
            return 0;
        }
    }
    cout << "Yes" << endl;

}

File: ABC183/D.py
import sys

sys.setrecursionlimit(10 ** 9)

N, W = map(int, sys.stdin.readline().strip().split())

cusum = [0] * (2 * 10 ** 5 + 1)

for n in range(N):
    s, t, p = list(map(int, sys.stdin.readline().strip().split()))
    cusum[s] += p
    cusum[t] -= p

if cusum[0] > W:
    print("No")
    sys.exit()

for i in range(len(cusum) - 1):
    cusum[i + 1] += cusum[i]
    if cusum[i + 1] > W:
        print("No")
        sys.exit()


print("Yes")
